go/mysql/conn_test.go

```diff
@@ -928,19 +990,28 @@ func TestInitDbAgainstWrongDbDoesNotDropConnection(t *testing.T) {
 }
 
 func TestConnectionErrorWhileWritingComQuery(t *testing.T) {
-	// Set the conn for the server connection to the simulated connection which always returns an error on writing
-	sConn := newConn(testConn{
-		writeToPass: []bool{false, true},
-		pos:         -1,
-		queryPacket: []byte{0x21, 0x00, 0x00, 0x00, ComQuery, 0x73, 0x65, 0x6c, 0x65, 0x63, 0x74, 0x20, 0x40, 0x40, 0x76, 0x65, 0x72, 0x73,
-			0x69, 0x6f, 0x6e, 0x5f, 0x63, 0x6f, 0x6d, 0x6d, 0x65, 0x6e, 0x74, 0x20, 0x6c, 0x69, 0x6d, 0x69, 0x74, 0x20, 0x31},
-	}, DefaultFlushDelay, 0)
-
-	// this handler will return an error on the first run, and fail the test if it's run more times
-	errorString := make([]byte, 17000)
-	handler := &testRun{err: errors.New(string(errorString))}
-	res := sConn.handleNextCommand(handler)
-	require.False(t, res, "we should beak the connection in case of error writing error packet")
+	origMysqlMultiQuery := mysqlMultiQuery
+	defer func() {
+		mysqlMultiQuery = origMysqlMultiQuery
+	}()
```

Comment:
we should continue to test old and new code paths.

Reply:
As the new code path is behind the flag, mostly old code path will be used in v22. On `main` we can remove it

Reply:
We are testing both code-paths. That's what the for loop and t.Run are for.